scipy/optimize/meson.build

```diff
@@ -0,0 +1,117 @@
+include_dirs = [incdir_numpy, '../_lib/src']
+
+minpack_lib = static_library('minpack', ['minpack/hybrj.f', 
+     'minpack/lmstr1.f', 
+     'minpack/lmdif.f', 
+     'minpack/fdjac1.f', 
+     'minpack/fdjac2.f', 
+     'minpack/enorm.f', 
+     'minpack/lmder1.f', 
+     'minpack/r1mpyq.f', 
+     'minpack/r1updt.f', 
+     'minpack/lmdif1.f', 
+     'minpack/rwupdt.f', 
+     'minpack/qform.f', 
+     'minpack/qrfac.f', 
+     'minpack/hybrd1.f', 
+     'minpack/lmder.f', 
+     'minpack/chkder.f', 
+     'minpack/qrsolv.f', 
+     'minpack/lmstr.f', 
+     'minpack/hybrj1.f', 
+     'minpack/dpmpar.f', 
+     'minpack/hybrd.f', 
+     'minpack/dogleg.f', 
+     'minpack/lmpar.f'])
+
+_minpack = py3.extension_module('_minpack',
+  ['minpack.h', '__minpack.h', '_minpackmodule.c'],
+  link_with : minpack_lib,
+  c_args : numpy_nodepr_api,
+  dependencies : [py3_dep],
+  include_directories : include_dirs,
+  install : true,
+  subdir : 'scipy/optimize')
+
+compiler = meson.get_compiler('cpp')
+compiler_id = compiler.get_id()
+std_flag = ''
+if compiler_id == 'msvc'
+  if compiler.has_argument('/std=c++14')
+    std_flag = '/std=c++14'
+  endif
+elif compiler_id == 'intelw'
+  if compiler.has_argument('/Qstd=c++14')
+    std_flag = '/Qstd=c++14'
+  elif compiler.has_argument('/Qstd=c++11')
+    std_flag = '/Qstd=c++11'
+  endif
+else
+  if compiler.has_argument('-std=c++14')
+    std_flag = '-std=c++14'
+  elif compiler.has_argument('-std=c++11')
+    std_flag = '-std=c++11'
+  endif
```

Comment:
This whole block should not be needed. The top-level `meson.build` has:
```
default_options: [
  'buildtype=debugoptimized',
  'c_std=c99',
  'cpp_std=c++14',  # TODO: use c++11 if 14 is not available
```
`cpp_std` is a builtin option so it doesn't need to be custom-handled, see https://mesonbuild.com/Builtin-options.html#compiler-options.

Reply:
I removed the flag handling and the build output is the same. So, resolving this. Thanks.